include/alpaka/vec/Vec.hpp

```diff
@@ -169,6 +169,8 @@ namespace alpaka
             typename = typename std::enable_if<
                 // There have to be dim arguments.
                 (sizeof...(TArgs)+1 == TDim::value)
+                &&
+                (std::is_same<TSize, TArg0>::value)
```

Comment:
This will not work if `TArg0` is a reference type. You should use `std::is_same<TSize, typename std::decay<TArg0>::type>::value` instead.
Furthermore, I think it would be better to replace the current line 176 with this line because the new line is just a stricter version of the current line 176.